apps/src/lab2/views/Lab2Wrapper.tsx

```diff
@@ -38,34 +39,25 @@ export interface Lab2WrapperProps {
   children: React.ReactNode;
 }
 
-async function fetchIsProjectValidator(): Promise<boolean> {
-  const permissionsResponse = await HttpClient.fetchJson<{
-    permissions: string[];
-  }>('/api/v1/users/current/permissions');
-  const {permissions} = permissionsResponse.value;
-
-  return permissions.includes(PERMISSIONS.PROJECT_VALIDATOR) ? true : false;
-}
-
 const Lab2Wrapper: React.FunctionComponent<Lab2WrapperProps> = ({children}) => {
   const isLoading: boolean = useSelector(isLabLoading);
   const isPageError: boolean = useSelector(hasPageError);
   const isBlocked = useAppSelector(state => state.lab.isBlocked);
   const [isProjectValidator, setIsProjectValidator] = useState(false);
+  const dispatch = useAppDispatch();
   useEffect(() => {
-    fetchIsProjectValidator().then(data => {
-      setIsProjectValidator(data);
+    fetchPermissions().then(data => {
+      dispatch(setPermissions(data));
+      setIsProjectValidator(data.includes(PERMISSIONS.PROJECT_VALIDATOR));
```

Comment:
Do we still need this stored as component state? Can this just be derived from the `permissions` in redux?